misaligned.cpp

```diff
@@ -1,21 +1,27 @@
 #include <assert.h>
 #include <iostream>
 
-int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
-    int i = 0, j = 0;
-    for (i = 0; i < 5; i++) {
-        for (j = 0; j < 5; j++) {
-            std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
-        }
+using std::cout;
+const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
+enum majorColors{White, Red, Black, Yellow, Violet};
+const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+
+int printColorMap(int pairNum, int majorNum) {
+    int totalPairs = 0;
+    int minorNum = 0;
+    for (int minorColorNum = 0; minorColorNum < 5; minorColorNum++) {
+        cout <<pairNum++ << " | " << majorColor[majorNum] << " | " << minorColor[minorNum] << "\n";
+        totalPairs++;
     }
-    return i * j;
+    return totalPairs;
 }
 
 int main() {
-    int result = printColorMap();
-    assert(result == 25);
-    std::cout << "All is well (maybe!)\n";
+    assert(printColorMap(0, White) == 5);
+    assert(printColorMap(5, Red) == 5);
+    assert(printColorMap(10, Black) == 5);
+    assert(printColorMap(15, Yellow) == 4);

```

Comment:
In this case, the test fails due to a mistake in the test itself. 
Keep the test integrity. Make it fail due to the mistake in the code.